fix: replace generic proof record type

heueristik opened the pull request:
@paulcadman can you review and see if this makes sense? Should we use the generic proof record type here as we are doing it already within the `Action` record?

https://github.com/anoma/juvix-arm-specs/blob/311f737f007e938b1b34c6cb2c738f0be9f41c5c/Anoma/Transaction/Action.juvix#L17-L23

paulcadman reviewed (commented):
I know we discussed this at some point but I can't remember the conclusion.

Q: Should the Delta `prove` function use the generic proof type. It's using GenericProofRecord which I thought we had removed. There seems to be some inconsistency in the types between how delta proofs are computed and how they're stored in the transaction.

https://github.com/anoma/juvix-arm-specs/blob/edf4548dc34108640cd784eb0b18d97b21286932/Anoma/Proving/DeltaProof.juvix#L10

heueristik commented:
> I know we discussed this at some point but I can't remember the conclusion.
> 
> Q: Should the Delta `prove` function use the generic proof type. It's using `Parametrized.GenericProofRecord` which I thought we had removed.

Oh, you are absolutely right. Seems like I forgot to remove it. I changed it in 97fca11ee21b00cf9c8b6273330bf61117ec5231. Is it correct now?


> There seems to be some inconsistency in the types between how delta proofs are computed and how they're stored in the transaction.
> https://github.com/anoma/juvix-arm-specs/blob/edf4548dc34108640cd784eb0b18d97b21286932/Anoma/Proving/DeltaProof.juvix#L10-L14

Yes, I think proof records are stored in the transaction like they are within an action. Maybe we/I should ask Yulia for clarification.
